azure-proto-core/Adapters/ModelAdapters.cs

```diff
@@ -24,4 +24,22 @@ protected TrackedResource(ResourceIdentifier id, Location location, T data)
             return other.Model;
         }
     }
+
+    //Or call generic resource, other resource??
+    public abstract class ChildResource<T> : Resource where T : class
```

Comment:
I think we should go with SubResource - this is the term used most often in Swagger